packages/loader/core-interfaces/src/index.ts

```diff
@@ -8,7 +8,16 @@
 export { IFluidObject } from "./fluidObject";
 
 export * from "./fluidLoadable";
-export * from "./fluidRouter";
+// Typescript forgets the index signature when customers augment IRequestHeader if we export *.
+// So we export the explicit members as a workaround:
+// https://github.com/microsoft/TypeScript/issues/18877#issuecomment-476921038
+export {
+    IRequest,
```

Comment:
I wonder whether we can just export by name the one sthat we would augment by name and the rest can still be export *?

Reply:
We can with some eslint-disables, either way seems fine to me:

```typescript
export {
    // eslint-disable-next-line import/export
    IRequestHeader,
} from "./fluidRouter";
// eslint-disable-next-line import/export
export * from "./fluidRouter";
```

Reply:
I probably prefer the current approach personally, but happy to change if you'd like.